src/main/java/de/tum/cit/aet/artemis/lecture/service/LectureUnitService.java

```diff
@@ -243,4 +243,15 @@ public URL validateUrlStringAndReturnUrl(String urlString) {
             throw new BadRequestException();
         }
     }
+
+    /**
+     * Ingest the lectureUnit when triggered by the ingest lectureUnit button
+     *
+     * @param lectureUnit lectureUnit to be ingested
+     */
+    public void ingestLectureUnitInPyris(LectureUnit lectureUnit) {
+        if (lectureUnit instanceof AttachmentUnit && pyrisWebhookService.isPresent()) {
+            pyrisWebhookService.get().addLectureUnitToPyrisDB((AttachmentUnit) lectureUnit);
```

Comment:
```suggestion
        if (lectureUnit instanceof AttachmentUnit attachmentUnit && pyrisWebhookService.isPresent()) {
            pyrisWebhookService.get().addLectureUnitToPyrisDB(attachmentUnit);
```

Reply:
Would it make sense to add an else here and at least log something? potentially you could also notify the client that the lecture unit cannot be ingested or that Pyris is currently not available

Reply:
I added a bad request and service unavailable responses for the client, for the logging part it is already happening in other methods, that way the logging is more helpful and specific